File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/core/ImageInfo.kt
package dev.hebei.camerax_android.legacy.core

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiImageInfo

class ImageInfo(registrar: CameraXRegistrar) : PigeonApiImageInfo(registrar) {
    override fun timestamp(pigeon_instance: androidx.camera.core.ImageInfo): Long {
        return pigeon_instance.timestamp
    }

    override fun rotationDegrees(pigeon_instance: androidx.camera.core.ImageInfo): Long {
        return pigeon_instance.rotationDegrees.toLong()
    }
}

File: camerax_android/lib/src/legacy/core/image_info.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension ImageInfoObj on $native.ImageInfo {
  $interface.ImageInfo get args {
    return $interface.ImageInfo(
      timestamp: timestamp,
      rotationDegrees: rotationDegrees,
    );
  }
}
